test/Dialect/FIRRTL/SFCTests/mem-taps-reg.fir

```diff
@@ -54,12 +54,12 @@ circuit Top : %[[
 
 ; CHECK:      module Top(
 ; CHECK-NOT:  module
-; CHECK:  assign memTap_0 = Top.dut.rf_0;
-; CHECK:  assign memTap_1 = Top.dut.rf_1;
-; CHECK:  assign memTap_2 = Top.dut.rf_2;
-; CHECK:  assign memTap_3 = Top.dut.rf_3;
-; CHECK:  assign memTap_4 = Top.dut.rf_4;
-; CHECK:  assign memTap_5 = Top.dut.rf_5;
-; CHECK:  assign memTap_6 = Top.dut.rf_6;
-; CHECK:  assign memTap_7 = Top.dut.rf_7;
+; CHECK:        wire [7:0] memTap_0 = Top.dut.rf_0;
```

Comment:
I hate how fragile this is.  I've also had to change it back and forth as it is so easy to alternate between decl + assign and inline assign.

Reply:
Yeah. I think we'll be able to make this more robust by topologically sorting the modules before emission now that `hw.wire` is a thing; that makes decl+assign vs decl+init depend solely on the structure of the SSA graph, not some accidental op ordering, which should be more stable and easier to predict.